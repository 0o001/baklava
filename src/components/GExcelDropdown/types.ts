import { GExcelDropdownStatus } from '@/components/GExcelDropdown/enums';

export interface GExcelDropdownListItem {
  status: GExcelDropdownStatus,
  fileName: string,
  downloadUrl: string,
}
