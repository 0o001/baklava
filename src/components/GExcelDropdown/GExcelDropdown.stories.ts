import { storiesOf } from '@storybook/vue';
import GExcelDropdown from './GExcelDropdown.vue';
import { action } from '@storybook/addon-actions';
import { number, withKnobs } from '@storybook/addon-knobs';

storiesOf('GExcelDropdown', module)
  .addParameters({ component: GExcelDropdown })
  .addDecorator(withKnobs)
  .add('Base', () => ({
    components: { GExcelDropdown },
    props: {
      historyIntervalMs: {
        default: number('historyIntervalMs', 15000),
      },
    },
    methods: {
      excelUpload (event) {
        action('excelUpload')();
      },
      excelUploadHistory () {
        action('excelUploadHistory')();
      },
      excelDownload (event) {
        action('excelDownload')();
      },
      excelTabType (event) {
        action('excelTabType')();
      },
      excelDownloadHistory () {
        action('excelDownloadHistory')();
      },
    },
    template: `
      <GExcelDropdown
        :download-tab-text="'Daha önce eklediğiniz excel bulunmamaktadır.'"
        :upload-tab-text="'Daha önce eklediğiniz excel bulunmamaktadır.'"
        :download-button-text="'EXCEL İNDİR'"
        :upload-button-text="'EXCEL YÜKLE'"
        :excel-history-list="[]"
        :history-interval-ms="historyIntervalMs"
        @excelTapType="excelTabType"
        @excelUpload="excelUpload"
        @excelUploadHistory="excelUploadHistory"
        @excelDownload="excelDownload"
        @excelDownloadHistory="excelDownloadHistory"
      />
    `,
  }))
  .add('Only Download', () => ({
    components: { GExcelDropdown },
    props: {
      historyIntervalMs: {
        default: number('historyIntervalMs', 15000),
      },
    },
    methods: {
      excelDownload (event) {
        action('excelDownload')();
      },
      excelDownloadHistory () {
        action('excelDownloadHistory')();
      },
    },
    template: `
      <GExcelDropdown
        :excel-history-list="[{
          status: 'CREATED',
          fileName: 'deneme',
          downloadUrl: 'deneme',
        }, {
          status: 'CREATED',
          fileName: 'deneme',
          downloadUrl: 'deneme',
        }, {
          status: 'FAILED',
          fileName: 'deneme',
          downloadUrl: 'deneme',
        }]"
        :history-interval-ms="historyIntervalMs"
        @excelDownload="excelDownload"
        @excelDownloadHistory="excelDownloadHistory"
        download
      />
    `,
  }))
  .add('With Upload And Download', () => ({
    components: { GExcelDropdown },
    props: {
      historyIntervalMs: {
        default: number('historyIntervalMs', 15000),
      },
    },
    methods: {
      excelUpload (event) {
        action('excelUpload')();
      },
      excelUploadHistory () {
        action('excelUploadHistory')();
      },
      excelDownload (event) {
        action('excelDownload')();
      },
      excelDownloadHistory () {
        action('excelDownloadHistory')();
      },
    },
    template: `
      <GExcelDropdown
        :excel-history-list="[{
          status: 'CREATED',
          fileName: 'deneme',
          downloadUrl: 'deneme',
        }, {
          status: 'CREATED',
          fileName: 'deneme',
          downloadUrl: 'deneme',
        }, {
          status: 'FAILED',
          fileName: 'deneme',
          downloadUrl: 'deneme',
        }]"
        :history-interval-ms="historyIntervalMs"
        @excelUpload="excelUpload"
        @excelUploadHistory="excelUploadHistory"
        @excelDownload="excelDownload"
        @excelDownloadHistory="excelDownloadHistory"
      />
    `,
  }));
