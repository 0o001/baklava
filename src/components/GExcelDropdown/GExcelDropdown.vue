<template>
  <div class="g-excel-dropdown">
    <GButtonDropdown
      variant="success"
      :is-dropdown-open="isDropDownOpen"
      @onDropdownOpen="onDropdownOpen"
      @onDropdownClose="onDropdownClose"
    >
      {{ buttonText }}
      <template #menu>
        <GTabs
          v-if="!download"
          :tabs="tabs"
          :type="type"
          @handleTabClick="handleTabClick"
        />
        <div class="excel-history">
          <div class="g-mb-16 title">
            <GText
              v-if="type === 'download'"
              variant="subtitle-03"
              color="green-500"
            >
              İNDİRME GEÇMİŞİM
            </GText>
            <GText
              v-else
              variant="subtitle-03"
              color="green-500"
            >
              YÜKLEME GEÇMİŞİM
            </GText>
          </div>
          <ul class="list">
            <li
              v-if="excelHistoryList.length === 0"
              @click="handleExcelDownload"
              class="list-item no-data"
            >
              <GText
                v-if="downloadTabText || uploadTabText"
                variant="menu-item"
              >
                <template v-if="type === 'download'">
                  {{ downloadTabText }}
                </template>
                <template v-else>
                  {{ uploadTabText }}
                </template>
              </GText>
            </li>
            <li
              v-else
              class="g-d-flex g-d-align-items-center list-item"
              v-for="(listItem, index) in excelHistoryList"
              :key="index"
            >
              <GIcon
                v-if="listItem.status === excelDropdownStatus.WAITING"
                name="circle"
                size="12px"
                ref="waitingSpinner"
              />
              <GIcon
                v-else
                :name="iconClass(listItem.status)"
                :stroke="iconColor(listItem.status)"
                size="12px"
                ref="fileIcon"
              />
              <GText
                variant="menu-item"
                :color="labelColor(listItem.status)"
                class="file-name"
                ref="fileName"
              >
                <a
                  v-if="listItem.status === excelDropdownStatus.CREATED"
                  :href="listItem.url"
                  :class="{ ready: listItem.status === excelDropdownStatus.CREATED }"
                  download
                >
                  {{ listItem.fileName }}
                </a>
                <span
                  v-else-if="listItem.status === excelDropdownStatus.WAITING"
                >
                  Hazırlanıyor...
                </span>
                <span
                  v-else-if="listItem.status === excelDropdownStatus.FAILED"
                >
                  Bir hata oluştu.
                </span>
              </GText>
            </li>
          </ul>
          <GButton
            v-if="type === 'download' && !download"
            variant="success"
            right-icon="download"
            class="g-mt-10"
            @click="handleExcelDownload"
            :disabled="isButtonDisabled()"
            ref="downloadButton"
            fluid
          >
            {{ downloadButtonText }}
          </GButton>
          <GButton
            v-else-if="!download"
            variant="success"
            right-icon="upload"
            class="g-mt-10 upload-button"
            @click="chooseFile"
            :disabled="isButtonDisabled()"
            ref="uploadButton"
            fluid
          >
            {{ uploadButtonText }}
          </GButton>
          <input
            @change="handleExcelUpload"
            id="fileUpload"
            type="file"
            hidden
          >
        </div>
      </template>
    </GButtonDropdown>
  </div>
</template>

<script lang="ts">

import GButtonDropdown from '../GButtonDropdown/GButtonDropdown.vue';
import GTabs from '../GTabs/GTabs.vue';
import GText from '../GText/GText.vue';
import GIcon from '../GIcon/GIcon.vue';
import GButton from '../GButton/GButton.vue';
import { GExcelDropdownStatus } from './enums.ts';

export default {
  name: 'GExcelDropdown',
  components: {
    GButtonDropdown,
    GTabs,
    GText,
    GIcon,
    GButton,
  },
  props: {
    download: {
      default: false,
      type: Boolean,
    },
    downloadTabText: {
      default: '',
      type: String,
    },
    uploadTabText: {
      default: '',
      type: String,
    },
    downloadButtonText: {
      default: 'EXCEL\'E AKTAR',
      type: String,
    },
    uploadButtonText: {
      default: 'EXCEL YÜKLE',
      type: String,
    },
    excelHistoryList: {
      default: () => [],
      type: Array,
    },
    historyIntervalMs: {
      default: 15000,
      type: Number,
    },
    buttonText: {
      default: 'EXCEL İŞLEMLERİ',
      type: String,
    },
    disabled: {
      default: false,
      type: Boolean,
    },
  },
  data () {
    return {
      isDropDownOpen: false,
      type: 'download',
      interval: null,
      tabs: [{
        type: 'download',
        label: 'İNDİRME İŞLEMLERİ',
      }, {
        type: 'upload',
        label: 'YÜKLEME İŞLEMLERİ',
      }],
    };
  },
  computed: {
    excelDropdownStatus () {
      return GExcelDropdownStatus;
    },
  },
  watch: {
    type (newValue) {
      if (this.interval) {
        clearInterval(this.interval);
      }

      this.type === 'download'
        ? this.setExcelDownloadHistoryInterval()
        : this.setExcelUploadHistoryInterval();
    },
  },
  methods: {
    iconClass (status: GExcelDropdownStatus): string {
      return {
        [GExcelDropdownStatus.CREATED]: 'check',
        [GExcelDropdownStatus.FAILED]: 'x',
      }[status];
    },
    iconColor (status: GExcelDropdownStatus): string {
      return status === GExcelDropdownStatus.CREATED ? 'var(--green-500)' : 'var(--red-500)';
    },
    labelColor (status: GExcelDropdownStatus): string {
      return status === GExcelDropdownStatus.CREATED ? 'main-grey-500' : 'mid-grey-500';
    },
    isButtonDisabled () {
      return this.excelHistoryList.some(listItem => listItem.status === this.excelDropdownStatus.WAITING);
    },
    handleTabClick (type: string): void {
      this.$emit('excelTabType', type);
      this.type = type;
    },
    onButtonClick (): void {
      this.isDropDownOpen = !this.isDropDownOpen;
    },
    onDropdownOpen (): void {
      this.type === 'download'
        ? this.setExcelDownloadHistoryInterval()
        : this.setExcelUploadHistoryInterval();
    },
    onDropdownClose (): void {
      clearInterval(this.interval);
    },
    chooseFile () {
      document.getElementById('fileUpload').click();
    },
    handleExcelUpload (event: any): void {
      const file = event.target.files[0];
      const data = new FormData();
      data.append('file', file);

      this.$emit('excelUpload', data);
    },
    handleExcelUploadHistory (): void {
      this.$emit('excelUploadHistory');
    },
    handleExcelDownload (): void {
      this.$emit('excelDownload');
    },
    handleExcelDownloadHistory (): void {
      this.$emit('excelDownloadHistory');
    },
    setExcelUploadHistoryInterval () {
      this.handleExcelUploadHistory();
      this.interval = setInterval(this.handleExcelUploadHistory, this.historyIntervalMs);
    },
    setExcelDownloadHistoryInterval () {
      this.handleExcelDownloadHistory();
      this.interval = setInterval(this.handleExcelDownloadHistory, this.historyIntervalMs);
    },
  },
};
</script>

<style lang="scss" scoped>
.g-excel-dropdown {
  ::v-deep .g-button-dropdown {
    .menu {
      padding: 0;
      z-index: 99;
      position: absolute;
      max-width: 385px;
    }
  }

  .excel-history {
    padding: 20px;

    .list {
      max-height: 240px;
      overflow-y: auto;
      padding: 0;
      list-style: none;

      .list-item {
        padding-top: var(--spacing-15);

        &:not(:last-child) {
          padding-bottom: var(--spacing-15);
          border-bottom: 1px solid var(--mid-grey-800);
        }

        a{
          white-space: nowrap;
          overflow: hidden;
          text-overflow: ellipsis;
          max-width: 320px;
          width: 320px;
          display: block;
        }

        .file-name {
          margin-left: var(--spacing-8);

          ::v-deep .ready {
            text-decoration: underline;
            cursor: pointer;
          }
        }

        &:last-child{
          padding-bottom: 20px;
        }
      }
    }
  }
}
</style>
