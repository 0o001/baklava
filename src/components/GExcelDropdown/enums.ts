export enum GExcelDropdownStatus {
  WAITING = 'WAITING_FOR_PROCESSED',
  CREATED = 'CREATED',
  FAILED = 'FAILED',
}
