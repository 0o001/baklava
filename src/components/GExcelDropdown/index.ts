import GExcelDropdown from './GExcelDropdown.vue';

export default GExcelDropdown;
