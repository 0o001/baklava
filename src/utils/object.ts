type AnyObject = { [key: string]: any }

export const isEmptyObject = (obj: AnyObject = {}) => {
  for (let i in obj) {
    return false;
  }

  return true;
};

export const isEqualObject = (a: AnyObject, b: AnyObject) => {
    return JSON.stringify(a) === JSON.stringify(b);
}
