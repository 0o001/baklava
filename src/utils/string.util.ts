const LEADING_WHITESPACE_REGEX = /^\s*/g;
const CONSECUTIVE_WHITESPACE_REGEX = /\s\s+/g;

const _removeLeadingSpaces = (string: string): string => {
  return string.replace(LEADING_WHITESPACE_REGEX, '');
};

const _removeConsecutiveSpaces = (string: string): string => {
  return string.replace(CONSECUTIVE_WHITESPACE_REGEX, ' ');
};

export const trimInputValue = (string: string): string => {
  if (typeof string !== 'string') return string;
  let tempString = string;
  tempString = _removeLeadingSpaces(tempString);
  tempString = _removeConsecutiveSpaces(tempString);
  return tempString;
};
