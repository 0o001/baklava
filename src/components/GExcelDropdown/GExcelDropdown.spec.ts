import { renderWrapper } from '@/utils/testUtils';
import { Wrapper } from '@vue/test-utils';
import GExcelDropdown from './GExcelDropdown.vue';

jest.useFakeTimers();

describe('GExcelDropdown Specs', () => {
  let wrapper: Wrapper<GExcelDropdown & { [key: string]: any }>;

  const excelHistoryList = [
    {
      status: 'CREATED',
      fileName: 'created file name',
      downloadUrl: 'created file url',
    },
    {
      status: 'WAITING_FOR_PROCESSED',
      fileName: 'waiting file name',
      downloadUrl: 'waiting file name',
    },
    {
      status: 'FAILED',
      fileName: 'failed file name',
      downloadUrl: 'failed file name',
    },
  ];

  const historyIntervalMsExp = 15001;

  afterEach(() => {
    jest.clearAllTimers();
  });

  it('should match snapshot with default props', () => {
    // given
    wrapper = renderWrapper(GExcelDropdown);

    // then
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should match snapshot with excel history list', () => {
    // given
    wrapper = renderWrapper(GExcelDropdown, { propsData: { excelHistoryList } });

    // then
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should emit excelTabType when clicked tab', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown);
    await wrapper.setData({ isDropDownOpen: true });
    // when
    await wrapper.get('.tab.upload').trigger('click');
    // then

    // @ts-ignore
    expect(wrapper.emitted('excelTabType').length).toBe(1);
  });

  it('should show excel dropdown menu and emit active tab history when clicked dropdown button', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown, {
      data () {
        return {
          type: 'download',
        };
      },
    });

    // when
    await wrapper.get('.g-button').trigger('click');

    // then
    expect(wrapper.get('.g-excel-dropdown .menu').classes()).not.toContain('d-none');
    // @ts-ignore
    expect(wrapper.emitted('excelDownloadHistory').length).toBe(1);

    jest.runTimersToTime(historyIntervalMsExp);
    // @ts-ignore
    expect(wrapper.emitted('excelDownloadHistory').length).toBe(2);
  });

  it('should emit excelUploadHistory when clicked upload operations tab', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown);
    await wrapper.setData({ isDropDownOpen: true, type: 'download' });

    // when
    await wrapper.get('.g-excel-dropdown .tab.upload').trigger('click');

    // then
    // @ts-ignore
    expect(wrapper.emitted('excelUploadHistory').length).toBe(1);

    jest.advanceTimersByTime(historyIntervalMsExp);
    // @ts-ignore
    expect(wrapper.emitted('excelUploadHistory').length).toBe(2);
  });

  it('should emit excelDownloadHistory when clicked upload operations tab', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown);
    await wrapper.setData({ isDropDownOpen: true, type: 'upload' });

    // when
    await wrapper.get('.g-excel-dropdown .tab.download').trigger('click');

    // then
    // @ts-ignore
    expect(wrapper.emitted('excelDownloadHistory').length).toBe(1);

    jest.advanceTimersByTime(historyIntervalMsExp);
    // @ts-ignore
    expect(wrapper.emitted('excelDownloadHistory').length).toBe(2);
  });

  it('should show no data information if excelHistoryList length equal zero', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown);
    await wrapper.setData({ isDropDownOpen: true });
    await wrapper.setProps({ downloadTabText: 'some text' });

    // then
    const noDataText = wrapper.get('.list-item.no-data');

    expect(noDataText.exists()).toBe(true);
    expect(noDataText.text()).toBe('some text');
  });

  it('should emit excelDownload when clicked downloadTabText', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown);
    await wrapper.setData({ isDropDownOpen: true });
    await wrapper.setProps({ downloadTabText: 'some text' });

    const noDataText = wrapper.get('.list-item.no-data');

    // when
    await noDataText.trigger('click');

    // then
    expect(wrapper.emitted('excelDownload')).toBeTruthy();
  });

  it('should emit excelDownload when clicked download button', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown);
    await wrapper.setData({ isDropDownOpen: true, type: 'download' });

    // when
    await wrapper.findComponent({ ref: 'downloadButton' }).trigger('click');

    // then
    expect(wrapper.emitted('excelDownload')).toBeTruthy();
  });

  it('should disabled excel download button if excel history has any waiting item', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown, { propsData: { excelHistoryList } });

    // when
    const downloadButton = wrapper.findComponent({ ref: 'downloadButton' });

    // then
    expect(downloadButton.attributes('disabled')).toBe('disabled');
  });

  it('should disabled excel upload button if excel history has any waiting item', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown, { propsData: { excelHistoryList } });
    await wrapper.setData({ isDropDownOpen: true, type: 'upload' });

    // when
    const uploadButton = wrapper.findComponent({ ref: 'uploadButton' });

    // then
    expect(uploadButton.attributes('disabled')).toBe('disabled');
  });

  it('should emit excelUpload with file data when called handleExcelUpload', async () => {
    // given
    const event = {
      target: {
        files: ['data'],
      },
    };

    wrapper = renderWrapper(GExcelDropdown);
    await wrapper.setData({ isDropDownOpen: true, type: 'upload' });

    // when
    wrapper.vm.handleExcelUpload(event);

    // then
    const data = new FormData();
    data.append('file', event.target.files[0]);

    // @ts-ignore
    expect(wrapper.emitted('excelUpload')[0]).toEqual([data]);
  });

  it('should render excel history list with expected icon and file names', async () => {
    // given
    wrapper = renderWrapper(GExcelDropdown, { propsData: { excelHistoryList } });
    await wrapper.setData({ isDropDownOpen: true, type: 'download' });

    // when
    const createdFileText = wrapper.findAllComponents({ ref: 'fileName' }).at(0);
    const waitingFileText = wrapper.findAllComponents({ ref: 'fileName' }).at(1);
    const failedFileText = wrapper.findAllComponents({ ref: 'fileName' }).at(2);

    const waitingSpinner = wrapper.findComponent({ ref: 'waitingSpinner' });
    const createdFileIcon = wrapper.findAllComponents({ ref: 'fileIcon' }).at(0);
    const failedFileIcon = wrapper.findAllComponents({ ref: 'fileIcon' }).at(1);

    // then
    expect(createdFileText.text()).toBe('created file name');
    expect(waitingFileText.text()).toBe('Hazırlanıyor...');
    expect(failedFileText.text()).toBe('Bir hata oluştu.');

    expect(waitingSpinner.exists()).toBeTruthy();
    expect(createdFileIcon.props('name')).toBe('check');
    expect(failedFileIcon.props('name')).toBe('x');
  });
});
