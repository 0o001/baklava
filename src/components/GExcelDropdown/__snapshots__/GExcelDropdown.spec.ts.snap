// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GExcelDropdown Specs should match snapshot with default props 1`] = `
<div
  class="g-excel-dropdown"
>
  <div
    class="g-button-dropdown"
  >
    <div
      class="buttons g-d-flex"
    >
      <button
        class="g-button -success"
      >
        <div
          class="content -big"
        >
          <!---->
           
    EXCEL İŞLEMLERİ
     
          <!---->
           
          <!---->
        </div>
      </button>
       
      <button
        class="dropdown-button g-button -icon -onlyIcon -success-darker"
      >
        <div
          class="content -big"
        >
          <!---->
            
          <svg
            class="g-icon feather feather--chevron-down"
            dataName="chevron-down"
            dataTags=""
            dataType="chevron-down"
            fill="none"
            height="20"
            stroke="var(--white)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
           
          <!---->
        </div>
      </button>
    </div>
     
    <div
      class="menu g-d-none"
    >
      <div
        class="g-tabs"
      >
        <div
          class="tab active download"
        >
          <div
            class="tab-title"
          >
            <div
              class="tab-name g-d-flex g-d-align-center g-d-justify-center g-text-main-grey-500 -subtitle-03 g-text"
            >
               
              <span
                class=""
              >
                
            İNDİRME İŞLEMLERİ
          
              </span>
               
              <!---->
               
              <!---->
            </div>
          </div>
           
          <!---->
           
          <div
            class="seperator"
          />
        </div>
        <div
          class="tab upload"
        >
          <div
            class="tab-title"
          >
            <div
              class="tab-name g-d-flex g-d-align-center g-d-justify-center g-text-main-grey-500 -subtitle-03 g-text"
            >
               
              <span
                class=""
              >
                
            YÜKLEME İŞLEMLERİ
          
              </span>
               
              <!---->
               
              <!---->
            </div>
          </div>
           
          <!---->
           
          <div
            class="seperator"
          />
        </div>
      </div>
       
      <div
        class="excel-history"
      >
        <div
          class="g-mb-16 title"
        >
          <div
            class="g-text-green-500 -subtitle-03 g-text"
          >
            
            İNDİRME GEÇMİŞİM
          
          </div>
        </div>
         
        <ul
          class="list"
        >
          <li
            class="list-item no-data"
          >
            <!---->
          </li>
        </ul>
         
        <button
          class="g-mt-10 g-button -fluid -icon -success"
        >
          <div
            class="content -big"
          >
            <!---->
             
          EXCEL'E AKTAR
         
            <!---->
             
            <svg
              class="-rightIcon g-icon feather feather--download g-ml-10"
              dataName="download"
              dataTags=""
              dataType="download"
              fill="none"
              height="20"
              stroke="var(--white)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
              />
              <polyline
                points="7 10 12 15 17 10"
              />
              <line
                x1="12"
                x2="12"
                y1="15"
                y2="3"
              />
            </svg>
          </div>
        </button>
         
        <input
          hidden="hidden"
          id="fileUpload"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`GExcelDropdown Specs should match snapshot with excel history list 1`] = `
<div
  class="g-excel-dropdown"
>
  <div
    class="g-button-dropdown"
  >
    <div
      class="buttons g-d-flex"
    >
      <button
        class="g-button -success"
      >
        <div
          class="content -big"
        >
          <!---->
           
    EXCEL İŞLEMLERİ
     
          <!---->
           
          <!---->
        </div>
      </button>
       
      <button
        class="dropdown-button g-button -icon -onlyIcon -success-darker"
      >
        <div
          class="content -big"
        >
          <!---->
            
          <svg
            class="g-icon feather feather--chevron-down"
            dataName="chevron-down"
            dataTags=""
            dataType="chevron-down"
            fill="none"
            height="20"
            stroke="var(--white)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
           
          <!---->
        </div>
      </button>
    </div>
     
    <div
      class="menu g-d-none"
    >
      <div
        class="g-tabs"
      >
        <div
          class="tab active download"
        >
          <div
            class="tab-title"
          >
            <div
              class="tab-name g-d-flex g-d-align-center g-d-justify-center g-text-main-grey-500 -subtitle-03 g-text"
            >
               
              <span
                class=""
              >
                
            İNDİRME İŞLEMLERİ
          
              </span>
               
              <!---->
               
              <!---->
            </div>
          </div>
           
          <!---->
           
          <div
            class="seperator"
          />
        </div>
        <div
          class="tab upload"
        >
          <div
            class="tab-title"
          >
            <div
              class="tab-name g-d-flex g-d-align-center g-d-justify-center g-text-main-grey-500 -subtitle-03 g-text"
            >
               
              <span
                class=""
              >
                
            YÜKLEME İŞLEMLERİ
          
              </span>
               
              <!---->
               
              <!---->
            </div>
          </div>
           
          <!---->
           
          <div
            class="seperator"
          />
        </div>
      </div>
       
      <div
        class="excel-history"
      >
        <div
          class="g-mb-16 title"
        >
          <div
            class="g-text-green-500 -subtitle-03 g-text"
          >
            
            İNDİRME GEÇMİŞİM
          
          </div>
        </div>
         
        <ul
          class="list"
        >
          <li
            class="g-d-flex g-d-align-items-center list-item"
          >
            <svg
              class="g-icon feather feather--check"
              dataName="check"
              dataTags=""
              dataType="check"
              fill="none"
              height="12px"
              stroke="var(--green-500)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="20 6 9 17 4 12"
              />
            </svg>
             
            <div
              class="file-name g-text-main-grey-500 -menu-item g-text"
            >
              <a
                class="ready"
                download=""
              >
                
                created file name
              
              </a>
            </div>
          </li>
          <li
            class="g-d-flex g-d-align-items-center list-item"
          >
            <svg
              class="g-icon feather feather--circle"
              dataName="circle"
              dataTags="off,zero,record"
              dataType="circle"
              fill="none"
              height="12px"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
            </svg>
             
            <div
              class="file-name g-text-mid-grey-500 -menu-item g-text"
            >
              <span>
                
                Hazırlanıyor...
              
              </span>
            </div>
          </li>
          <li
            class="g-d-flex g-d-align-items-center list-item"
          >
            <svg
              class="g-icon feather feather--x"
              dataName="x"
              dataTags="cancel,close,delete,remove,times,clear"
              dataType="x"
              fill="none"
              height="12px"
              stroke="var(--red-500)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="18"
                x2="6"
                y1="6"
                y2="18"
              />
              <line
                x1="6"
                x2="18"
                y1="6"
                y2="18"
              />
            </svg>
             
            <div
              class="file-name g-text-mid-grey-500 -menu-item g-text"
            >
              <span>
                
                Bir hata oluştu.
              
              </span>
            </div>
          </li>
        </ul>
         
        <button
          class="g-mt-10 g-button -fluid -icon -success"
          disabled="disabled"
        >
          <div
            class="content -big"
          >
            <!---->
             
          EXCEL'E AKTAR
         
            <!---->
             
            <svg
              class="-rightIcon g-icon feather feather--download g-ml-10"
              dataName="download"
              dataTags=""
              dataType="download"
              fill="none"
              height="20"
              stroke="var(--white)"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
              />
              <polyline
                points="7 10 12 15 17 10"
              />
              <line
                x1="12"
                x2="12"
                y1="15"
                y2="3"
              />
            </svg>
          </div>
        </button>
         
        <input
          hidden="hidden"
          id="fileUpload"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;
